### ISSUE TITLE
DAL is re-ordering items with the ModelSelect2Multiple widget

### ISSUE THREAD

**Issue by YasirKusay** [yourlabs/django-autocomplete-light#1357]
I have just noticed something interesting with DAL if you are using the ModelSelect2Multiple widget for the forms.ModelMultipleChoiceField. I initially selected apricot, then apples:
<img width="484" alt="Screen Shot 2024-03-22 at 7 08 07 pm" src="https://github.com/yourlabs/django-autocomplete-light/assets/59812220/70ec8927-d0bd-4144-83f5-2bc48c61633d">

I then deleted both and then selected apples, then apricot:

<img width="392" alt="Screen Shot 2024-03-22 at 7 08 52 pm" src="https://github.com/yourlabs/django-autocomplete-light/assets/59812220/889bc5b8-f317-4e11-8126-cc1c0e3a9d89">

However, once I selected apricot, the options selected above are re-ordered to apricot, then apples:

<img width="205" alt="Screen Shot 2024-03-22 at 7 09 56 pm" src="https://github.com/yourlabs/django-autocomplete-light/assets/59812220/d38e0ef6-99d6-4b77-a366-31a996d104b7">

In other words, they are re-ordered to the original order in image 1. It appears that there is something in the background that is preserving the original order and ordering the items here based on that. I do not want that and I want to preserve the order of the currently selected items regardless. Is that possible?

**Comment by YasirKusay** [comment]
Here is some code btw:

```
## forms.py

from dal import autocomplete
from django import forms
from .models import Food, Analysis
from dal import autocomplete

class AnalysisForm(forms.Form):
    class Meta:
        model = Analysis
        fields = ('__all__')
        
    food = forms.ModelMultipleChoiceField(queryset=Food.objects.all(),
                                     required=True,
                                     widget=autocomplete.ModelSelect2Multiple(url='test2-autocomplete',
                                                         attrs={'data-placeholder': 'Foods ...',
                                                                'data-minimum-input-length': 1.,
                                                                "class": "food-select-form",
                                                                }))
```

```
## models.py

from django.db import models

# Create your models here.
class Food(models.Model):
    name = models.CharField(primary_key=True, max_length=40)

    def __str__(self):
        return self.name
    
class City(models.Model):
    name = models.CharField(primary_key=True, max_length=40)

    def __str__(self):
        return self.name
    
class Analysis(models.Model):
    food = models.ForeignKey(Food, on_delete=models.CASCADE)
    cities = models.ManyToManyField(City, blank=True)
```

```
## views.py

from django.shortcuts import render
from .models import Food
from dal import autocomplete
from .forms import AnalysisForm

def test(request):
    form = AnalysisForm()
    print("A")
    return render(request, 'test2/test2_file.html', {'form': form})

class FoodAutocomplete(autocomplete.Select2QuerySetView):
    def get_queryset(self):
        qs = Food.objects.all()
        print(qs)

        if self.q:
            qs = qs.filter(name__istartswith=self.q)

        return qs
```
```
## test2_file.html

   <body>
        <h1>ASD</h1>
        <h1>Page Title</h1>
        <div class="container column">
            <form>
                <div class="all_autocomplete_forms column">
                    <div class="wrapperDiv">
                        {% csrf_token %}
                        {{ form.food }}
                    </div>
                </div>
                <button id="add_more" class="btn btn-secondary" type="button">Add more</button>
            </form>
        </div>
    </body>
```

**Comment by PurpleToti** [comment]
Fixed in yourlabs/django-autocomplete-light via PR #1408 (bug fixes, CI overhaul), merged 2026-04-21. `ModelSelect2Multiple.filter_choices_to_render` now preserves Select2 submission order via `Case`/`When`, and the empty-selection regression (queryset left unfiltered) is fixed — it now returns `.none()`.
